pkg/kubelet/dockertools/manager.go

```diff
@@ -1497,7 +1499,7 @@ func (dm *DockerManager) createPodInfraContainer(pod *api.Pod) (kubeletTypes.Doc
 		return "", err
 	}
 
-	id, err := dm.runContainerInPod(pod, container, netNamespace, "")
+	id, err := dm.runContainerInPod(pod, container, netNamespace, "", "")
```

Comment:
Added pid mode to infra container and moved the logic produce the pid mode string to a method and corresponding test case since it is used in two spots now.